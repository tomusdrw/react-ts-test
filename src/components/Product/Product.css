.product {
  padding: 15px;
  margin: 20px;
  background: #ddd;
  min-width: 200px;
  max-width: calc(33% - 70px);
}

.product.special {
  background: #eea;
}

.product img {
  display: block;
  width: 100%;
}

.product a { 
  color: #44b;
  cursor: pointer;
}
