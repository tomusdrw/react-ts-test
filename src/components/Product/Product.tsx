import * as React from 'react';
import './Product.css';

export interface ProductType {
  id: number;
  name: string;
  price: number;
  description: string;
}

export const Product = ({ id, name, price, description }: ProductType) => (
  <div className="product">
    <img src={`https://xpla.org/ext/lorempixel/250/250/technics/${id}`} />
    <h3>{name}</h3>
    <p>{description}</p>
    <h4 className="price">${(price / 100).toFixed(2)} z≈Ç</h4>
  </div>
);
