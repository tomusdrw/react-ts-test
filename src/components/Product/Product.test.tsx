import * as React from 'react';
import { Product } from './Product';

import * as renderer from 'react-test-renderer';

const product = {
  id: 5,
  name: 'Product 1',
  description: 'Lorem ipsum',
  price: 1000,
  isSpecial: false
};

it('renders correctly', () => {
  expect(renderer.create(
    <Product {...product} />
  ).toJSON()).toMatchSnapshot();
});
