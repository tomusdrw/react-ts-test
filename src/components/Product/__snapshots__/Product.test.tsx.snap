// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="product "
>
  <img
    src="https://xpla.org/ext/lorempixel/250/250/technics/5"
  />
  <h3>
    Product 1
  </h3>
  <p>
    Lorem ipsum
    <a
      onClick={[Function]}
    >
       more...
    </a>
  </p>
  <h4
    className="price"
  >
    10.00
     z≈Ç
  </h4>
</div>
`;
